gammapy/stats/variability.py

```diff
@@ -346,7 +349,15 @@ def TimmerKonig_lightcurve_simulator(
 
     random_state = get_random_state(random_state)
 
-    frequencies = np.fft.fftfreq(npoints, spacing.value)
+    if leakage_protection < 1.0:
+        raise ValueError("The leakage protection factor must be at least 1.0")
+
+    npoints_extended = int(npoints * leakage_protection)
+    if leakage_protection > 1.0 and npoints_extended == npoints:
+        raise Warning(
+            "The extended number of points is the same length as the desired number"
```

Comment:
Is there anything more that could be added here i.e when would this occur and how to fix that?

Reply:
Good point @Astro-Kirsty. The if condition is True when the leakage_protection is too small. User should increase it or we just add 1 new point instead of raising an error;)

Reply:
I don't think raising an error or forcing a new point would be the correct thing to do. 
Let's say that some user has an automated tool which returns that the optimal factor is 1.01 or something similar, which then for short enough series results in the same amount of points. This in general means that the original algorithm would not result in noise leakage (typical factors could be 5-100 if there is leakage). In this case I would simply throw the warning up, so that the user is aware and can either increase the factor himself or leave as is because he is confident that there is no leakage.
I could add a more detailed warning description

Reply:
If you choose to raise the warning, then I suggest to improve the message a little bit.